src/state_machine.zig

```diff
@@ -2234,6 +2248,15 @@ pub fn StateMachineType(
                 break :expires_at expires_at;
             };
 
+            // The only movement allowed in a closed account is
+            // posting or voiding the closing transfer:
+            if (dr_account.flags.closed and !p.flags.closing_debit_account) {
+                return .debit_account_closed;
+            }
+            if (cr_account.flags.closed and !p.flags.closing_credit_account) {
+                return .credit_account_closed;
+            }
```

Comment:
Ok, I'm also concerned by the "footguns" that can be caused by closing an account with balance (especially pending balance). But the orthogonality of primitives seems to outweigh this risk (e.g. it's the application policy to balance the account before closing it).

>X is posted?
  X is voided?

A closed account will not accept any transfer except by posting/voiding the same transfer that set the closed flag. In this case, the pending balance will not be reversible.

> X expires?

Expiration reverts the balance without a transfer, so it will succeed even for closed accounts.

(...)

Any thoughts on this?